# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaDecoder.h
# Commit: eed72505cf9b
# Full Hash: eed72505cf9b3c968aced5a6488f2660e1f02da4
# Author: Alex Chronopoulos <achronop@gmail.com>
# Date: 2019-08-06 16:15:05
# Regressor Bug: 1530996
# File Overlap Count: 1
# Description:
#   Bug 1530996 - Use the new storage mechanism to save the current playback benchmark. r=jya
#   
#   Create an event in MediaFormatReader the will signal to the HTMLMediaElement to initiate a new storing according to the latest VideoInfo. Also when the application is shutting down, trigger a new storing early enough, before all the events are disconnected.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D38316
# ==============================================================================

diff -r 426b558f2f43 -r eed72505cf9b dom/media/MediaDecoder.h
--- a/dom/media/MediaDecoder.h	Tue Aug 06 09:24:40 2019 +0000
+++ b/dom/media/MediaDecoder.h	Tue Aug 06 09:25:45 2019 +0000
@@ -42,6 +42,7 @@
 
 class AbstractThread;
 class DOMMediaStream;
+class DecoderBenchmark;
 class FrameStatistics;
 class VideoFrameContainer;
 class MediaFormatReader;
@@ -475,6 +476,8 @@
 
   void OnNextFrameStatus(MediaDecoderOwner::NextFrameStatus);
 
+  void OnStoreDecoderBenchmark(const VideoInfo& aInfo);
+
   void FinishShutdown();
 
   void ConnectMirrors(MediaDecoderStateMachine* aObject);
@@ -517,6 +520,9 @@
   // Counters related to decode and presentation of frames.
   const RefPtr<FrameStatistics> mFrameStats;
 
+  // Store a benchmark of the decoder based on FrameStatistics.
+  RefPtr<DecoderBenchmark> mDecoderBenchmark;
+
   RefPtr<VideoFrameContainer> mVideoFrameContainer;
 
   // True if the decoder has been directed to minimize its preroll before
@@ -576,6 +582,7 @@
   MediaEventListener mOnWaitingForKey;
   MediaEventListener mOnDecodeWarning;
   MediaEventListener mOnNextFrameStatus;
+  MediaEventListener mOnStoreDecoderBenchmark;
 
   // True if we have suspended video decoding.
   bool mIsVideoDecodingSuspended = false;